ItemList.py

```diff
@@ -1,283 +1,316 @@
-    'Compass (Bottom of the Well)':                    ('Compass',  None,  0xA3, None),
-    'Compass (Ice Cavern)':                            ('Compass',  None,  0xA4, None),
-    'Map (Deku Tree)':                                 ('Map',      None,  0xA5, None),
-    'Map (Dodongos Cavern)':                           ('Map',      None,  0xA6, None),
-    'Map (Jabu Jabus Belly)':                          ('Map',      None,  0xA7, None),
-    'Map (Forest Temple)':                             ('Map',      None,  0xA8, None),
-    'Map (Fire Temple)':                               ('Map',      None,  0xA9, None),
-    'Map (Water Temple)':                              ('Map',      None,  0xAA, None),
-    'Map (Spirit Temple)':                             ('Map',      None,  0xAB, None),
-    'Map (Shadow Temple)':                             ('Map',      None,  0xAC, None),
-    'Map (Bottom of the Well)':                        ('Map',      None,  0xAD, None),
-    'Map (Ice Cavern)':                                ('Map',      None,  0xAE, None),
-    'Small Key (Forest Temple)':                       ('SmallKey', True,  0xAF, {'progressive': float('Inf')}),
-    'Small Key (Fire Temple)':                         ('SmallKey', True,  0xB0, {'progressive': float('Inf')}),
-    'Small Key (Water Temple)':                        ('SmallKey', True,  0xB1, {'progressive': float('Inf')}),
-    'Small Key (Spirit Temple)':                       ('SmallKey', True,  0xB2, {'progressive': float('Inf')}),
-    'Small Key (Shadow Temple)':                       ('SmallKey', True,  0xB3, {'progressive': float('Inf')}),
-    'Small Key (Bottom of the Well)':                  ('SmallKey', True,  0xB4, {'progressive': float('Inf')}),
-    'Small Key (Gerudo Training Grounds)':             ('SmallKey',True, 0xB5, {'progressive': float('Inf')}),
-    'Small Key (Gerudo Fortress)':                     ('FortressSmallKey',True, 0xB6, {'progressive': float('Inf')}),
-    'Small Key (Ganons Castle)':                       ('SmallKey', True,  0xB7, {'progressive': float('Inf')}),
-    'Double Defense':                                  ('Item',     True,  0xB8, None),
-    'Magic Bean Pack':                                 ('Item',     True,  0xC9, None),
-    'Triforce Piece':                                  ('Item',     True,  0xCA, {'progressive': float('Inf')}),
-    'Zeldas Letter':                                   ('Item',     True,  0x0B, None),
-    'Time Travel':                                     ('Event',    True,  None, None),
-    'Scarecrow Song':                                  ('Event',    True,  None, None),
-    'Triforce':                                        ('Event',    True,  None, None),
+    'Bombs (5)':                                       ('Item',     None,  True,  0x01, None),
+    'Deku Nuts (5)':                                   ('Repeat',   True,  True,  0x02, {'limit_item': 'Deku Nuts (5) (Limited)'}),
```

Comment:
Repeatable nuts, sticks, and shields are of type repeat so that they aren't counted as major items but maybe they could be in which case they could just be 'Item'. Repeatable 'False' items are items that aren't supposed to be repeatable but if you did get it from the same location twice for some reason it wouldn't be any different from getting it once. Repeatable 'None' items are ones that you should never be able to get from the same location twice.

Reply:
Yeah, I think I'd rather have a descriptive type like 'Refill'.

So, logic-wise, there's no difference between repeatable 'False' and 'None'... Ah, there's a key above the table, maybe I missed that. Ok.

Reply:
'Refill' is already a type.